apps/daimo-mobile/src/view/shared/text.tsx

```diff
@@ -1,53 +1,79 @@
 import Octicons from "@expo/vector-icons/Octicons";
 import { ReactNode, useMemo } from "react";
-import { Text, TextProps, TextStyle } from "react-native";
+import { Text, TextProps } from "react-native";
 
 import { color, ss } from "./style";
 
 export const MAX_FONT_SIZE_MULTIPLIER = 1.4;
 
-function useStyle(baseStyle: TextStyle, { color }: { color?: string }) {
-  return useMemo(() => [baseStyle, { color }], [baseStyle, color]);
-}
-
-export const DaimoText = (props: TextProps) => {
-  return <Text {...props} maxFontSizeMultiplier={MAX_FONT_SIZE_MULTIPLIER} />;
+type TypographyProps = TextProps & {
+  variant?:
+    | "h1"
+    | "h2"
+    | "h3"
+    | "body"
+    | "bodyCaps"
+    | "bodyMedium"
+    | "metadata"
+    | "metadataLight"
+    | "para"
+    | "btnCaps"
+    | "error"
+    | "center"
+    | "emphasizedSmallText";
+  color?: string;
 };
 
-export function TextH1(props: TextProps & { color?: string }) {
-  return <DaimoText {...props} style={useStyle(ss.text.h1, props)} />;
+// Temporary component for text management pending a full refactor
+export function DaimoText({
```

Comment:
I extended this component to make it easier to handle typography presets through a single component (and also to allow easy passthrough of props to the underlying `Text` components for very specific edge cases). In the future, I plan to add more props for more fine-grained control of weight, color and other text properties (and possibly also do the same for the `Button` component).

I can also remove this if the previous version is preferred.